query/src/sql/planner/binder/join.rs

```diff
@@ -105,15 +105,27 @@ impl<'a> Binder {
                 left_child,
                 right_child,
             ),
-            JoinOperator::LeftOuter => Err(ErrorCode::UnImplement(
-                "Unsupported join type: LEFT OUTER JOIN",
-            )),
-            JoinOperator::RightOuter => Err(ErrorCode::UnImplement(
-                "Unsupported join type: RIGHT OUTER JOIN",
-            )),
-            JoinOperator::FullOuter => Err(ErrorCode::UnImplement(
-                "Unsupported join type: FULL OUTER JOIN",
-            )),
+            JoinOperator::LeftOuter => self.bind_join_with_type(
+                JoinType::Left,
+                left_join_conditions,
+                right_join_conditions,
+                left_child,
+                right_child,
+            ),
+            JoinOperator::RightOuter => self.bind_join_with_type(
+                JoinType::Left,
+                right_join_conditions,
+                left_join_conditions,
+                right_child,
+                left_child,
+            ),
+            JoinOperator::FullOuter => self.bind_join_with_type(
+                JoinType::Full,
+                left_join_conditions,
+                right_join_conditions,
+                left_child,
+                right_child,
+            ),
```

Comment:
Seems the full outer join hasn't been implemented yet?

Reply:
Yes

Reply:
Maybe we should return an error here, otherwise it will panic in `probe_join` if user writes down a `FULL OUTER JOIN`.